Automattic/jetpack
# My Jetpack: fix highlights and purchases API calls when not connected

Fixes:
- https://github.com/Automattic/jetpack/issues/37935

## Proposed changes:

* When the Jetpack connection isn't active we avoid performing some API calls.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Create a JN site
* Add the Jetpack Beta plugin and apply this branch.
* Open Chrome's network tab and make sure there are no request errors on the main Jetpack menu (My Jetpack).
* Apply the changes to a Simple site, go to the Jetpack menu, and make sure there are no regressions.
* Apply these changes on an Atomic site and make sure everything works as expected.


https://github.com/user-attachments/assets/f529565b-a052-4371-91bc-80cd6f4aa0d5
